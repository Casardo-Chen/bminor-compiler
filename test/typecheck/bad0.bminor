x: integer = 5;
y: float = 10;

main: function void ()=
{
	x: integer = y * 10 + 15 - 4;
}