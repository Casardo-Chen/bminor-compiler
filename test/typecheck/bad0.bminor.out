y resolves to global y
type error: declaration of float for y does not match expression type integer
type error: cannot multiply integer (10) to float (y)
type error: cannot add integer (15) to invalid type kind
